### ISSUE TITLE
deep app link tab-hold should have entry "open in app" 

### ISSUE THREAD

**Issue by Djfe** [mozilla-mobile/fenix#4577]
### Why/User Benefit/User Problem
@nahuhh had a good suggestion after #3330 was closed:
> When you tap-hold on a link: open in private tab, new tab, copy link, share link are available. Should also have an "open in..." menu listing

in-case #4575 gets implemented some app links might not open in external apps by default anymore.
So offering the manual option in the context menu makes sense.
Also it allows to visually distinguish app links when tab-holding

"Open in..." could show the dialogue to choose a different app from the default one:
Direct click opens in YouTube, but tab-hold -> open in -> allows to choose an alternative YouTube app this once

### Acceptance Criteria (how do I know when I’m done?)
open in... entry in the context-menu for app links

**Comment by NotWoods** [comment]
Duplicate of #1754
